### ISSUE TITLE
Loss with NaN by using self-defined model

### ISSUE THREAD

**Issue by alanyannick** [CSAILVision/semantic-segmentation-pytorch#59]
Thanks for your clean and flexible frameworks :)
I met a question "Loss appears NaN" when I changed the decoder to self-defined model structure， even though I used the same operation "log_softmax + NLLloss" to avoid the unstable probability with CrossEntropyLoss. 
Did I miss something? Looking forward to your response.

**Comment by hangzhaomit** [comment]
Try with a smaller learning rate, or even zero, to see if your model is correctly defined.

**Comment by alanyannick** [comment]
Thanks a lot! After modifying my learning rate to 1e-7, it works.
 It is really important to evaluate the suitable learning rate at the beginning of the training process.

**Comment by hangzhaomit** [comment]
Cool, good to know. The down-sampling rate problem is answered here.
#62
